### DISCUSSION TITLE
Test region R2 value incorrect

### DISCUSSION THREAD

**Discussion by AruunnAnanda** [google/lightweight_mmm#130] [Q&A] [answered]
Hi there,

Out of suspicion I went on to replicate the R2 value that appears on **top of True and predicted graph**. And, I noticed that I received different value by using sklearn library. 

Below is the graph from Google LMMM. It says R2 of 0.955. 

![image](https://user-images.githubusercontent.com/107944288/210249663-2f67e853-04b6-44d7-93a4-a986904d537d.png)

I received a R2 of 0.50 for the same data using sklearn library. Can you please clarify here?

![image](https://user-images.githubusercontent.com/107944288/210249921-7c9f370a-313e-41dc-b1b8-1f51d615c268.png)

Thanks,
Arun

**Comment by michevan** [comment, marked as answer]
thanks for flagging this! What's going on here is that LMMM uses the "Bayesian R-squared" which is defined slightly differently from the standard R-squared used by scikit-learn. The code for the Bayesian R-squared can be seen [here](https://python.arviz.org/en/stable/_modules/arviz/stats/stats.html#r2_samples), and here's a quick [preprint](http://www.stat.columbia.edu/~gelman/research/unpublished/bayes_R2.pdf) on the difference between the two. Feel free to use either version of R-squared for your work!
